lib/portage/package/ebuild/doebuild.py

```diff
@@ -2133,9 +2185,48 @@ def spawn(
         logname_backup = mysettings.configdict["env"].get("LOGNAME")
         mysettings.configdict["env"]["LOGNAME"] = logname
 
+    eapi = mysettings["EAPI"]
+
+    unexported_env_vars = None
+    if "export-pms-vars" not in mysettings.features or not eapi_exports_pms_vars(eapi):
+        unexported_env_vars = _unexported_pms_vars
+
+    if unexported_env_vars:
+        orig_env = mysettings.environ()
+        # Copy since we are potentially removing keys from the dict.
+        env = orig_env.copy()
+
+        t = env["T"]
+        if os.path.isdir(t):
```

Comment:
@zmedico is there a way to determine if we are in a phase that has the parent directory of T locked? I fear that `os.path.isdir(t)` is not robust, as it could` also simply return true if the directory happens to exist

Reply:
Maybe we should

```python
if phase not in _global_pid_phases and os.path.isdir(t)
```

not sure if the `and os.path.isdir(t)` is even required then

Reply:
Yep, that seems to work and do what we want